ui/src/cloud/components/OrgSettings.tsx

```diff
@@ -1,34 +1,56 @@
 // Libraries
-import {PureComponent} from 'react'
+import {FunctionComponent, useEffect, useState} from 'react'
 import {connect} from 'react-redux'
+import {get} from 'lodash'
 
 // Constants
 import {CLOUD} from 'src/shared/constants'
 
+// Types
+import {AppState, Organization} from 'src/types'
+
 // Actions
 import {getOrgSettings as getOrgSettingsAction} from 'src/cloud/actions/orgsettings'
 
+interface PassedInProps {
+  children: React.ReactElement<any>
+}
+interface StateProps {
+  org: Organization
+}
 interface DispatchProps {
   getOrgSettings: typeof getOrgSettingsAction
 }
 
-class OrgSettings extends PureComponent<DispatchProps> {
-  public componentDidMount() {
-    if (CLOUD) {
-      this.props.getOrgSettings()
+type Props = StateProps & DispatchProps & PassedInProps
+
+const OrgSettings: FunctionComponent<Props> = ({
+  org,
+  getOrgSettings,
+  children,
+}) => {
+  const [hasFetchedOrgSettings, setHasFetchedOrgSettings] = useState<boolean>(
+    false
+  )
+  useEffect(() => {
+    if (CLOUD && org && !hasFetchedOrgSettings) {
```

Comment:
just flying through this, but shouldn't we be reading this off of the redux RemoteDataState?

Reply:
I tried that. RemoteDataState is `Done` before there is an org object. So that doesn't work for me.

Reply:
Specifically, `resources.orgs.status` is the RemoteDataState for all orgs. A specific org doesn't have a RemoteDataState. And because I need the `id` off of a specific org, I can't rely on `resources.orgs.status` as the signal.

Reply:
lame